fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/BIOMonad3.scala

```diff
@@ -3,6 +3,12 @@ package izumi.functional.bio
 trait BIOMonad3[F[-_, +_, +_]] extends BIOApplicative3[F] {
   def flatMap[R, E, A, R2 <: R, E2 >: E, B](r: F[R, E, A])(f: A => F[R2, E2, B]): F[R2, E2, B]
   def flatten[R, E, A](r: F[R, E, F[R, E, A]]): F[R, E, A] = flatMap(r)(identity)
+  
+  @inline final def tailRecM[R, E, A, B](a: A)(f: A => F[R, E, Either[A, B]]): F[R, E, B] =
```

Comment:
@Adriani277 Should be non-final since BIOMonad does not yet imply suspension and can be implemented by a strict data structure (such as List) that would need a custom implementation